Index: git/webrtc/modules/video_coding/codecs/vp8/vp8_impl.cc
===================================================================
--- git.orig/webrtc/modules/video_coding/codecs/vp8/vp8_impl.cc
+++ git/webrtc/modules/video_coding/codecs/vp8/vp8_impl.cc
@@ -1108,8 +1108,7 @@ int VP8DecoderImpl::InitDecode(const Vid
   cfg.h = cfg.w = 0;  // set after decode
 
   vpx_codec_flags_t flags = 0;
-#if !defined(WEBRTC_ARCH_ARM) && !defined(WEBRTC_ARCH_ARM64) && \
-  !defined(ANDROID)
+#if 0
   flags = VPX_CODEC_USE_POSTPROC;
 #ifdef INDEPENDENT_PARTITIONS
   flags |= VPX_CODEC_USE_INPUT_PARTITION;
